src/analyzing/AST_to_IL.ml

```diff
@@ -1236,55 +1240,82 @@ and mk_switch_break_label env tok =
   in
   (break_label, [ mk_s (Label break_label) ], switch_env)
 
+and implicit_return env eorig tok =
+  (* We always expect a value from an expression that is implicitly
+   * returned, so void is set to false here.
+   *)
+  let ss, e = expr_with_pre_stmts ~void:false env eorig in
+  let ret = mk_s (Return (tok, e)) in
+  ss @ [ ret ]
+
+and expr_stmt env (eorig : G.expr) tok : IL.stmt list =
+  (* optimize? pass context to expr when no need for return value? *)
+  let ss, e = expr_with_pre_stmts ~void:true env eorig in
+
+  (* ExprStmts don't care about results of an expression, so it is rightly
```

Comment:
Maybe we should just not create an `mk_aux_var` when something returns `()` ?

Reply:
It works!